cli/cmd/proxy/stats/command.go

```diff
@@ -0,0 +1,237 @@
+package stats
+
+import (
+	"errors"
+	"fmt"
+	"github.com/hashicorp/consul-k8s/cli/common"
+	"github.com/hashicorp/consul-k8s/cli/common/flag"
+	"github.com/hashicorp/consul-k8s/cli/common/terminal"
+	"github.com/hashicorp/consul-k8s/cli/helm"
+	helmCLI "helm.sh/helm/v3/pkg/cli"
+	"io"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+const envoyAdminPort = 19000
+
+type StatsCommand struct {
+	*common.BaseCommand
+
+	helmActionsRunner helm.HelmActionsRunner
+
+	kubernetes kubernetes.Interface
+
+	restConfig *rest.Config
+
+	set *flag.Sets
+
+	flagKubeConfig  string
+	flagKubeContext string
+	flagNamespace   string
+	flagPod         string
+
+	once sync.Once
+	help string
+}
+
+func (c *StatsCommand) init() {
+	c.set = flag.NewSets()
+	if c.helmActionsRunner == nil {
+		c.helmActionsRunner = &helm.ActionRunner{}
+	}
+
+	f := c.set.NewSet("Command Options")
+	f.StringVar(&flag.StringVar{
+		Name:    "kubeconfig",
+		Aliases: []string{"c"},
+		Target:  &c.flagKubeConfig,
+		Default: "",
+		Usage:   "Path to kubeconfig file.",
+	})
+	f.StringVar(&flag.StringVar{
+		Name:    "context",
+		Target:  &c.flagKubeContext,
+		Default: "",
+		Usage:   "Kubernetes context to use.",
+	})
+	f.StringVar(&flag.StringVar{
+		Name:    "namespace",
+		Target:  &c.flagNamespace,
+		Usage:   "The namespace where the target Pod can be found.",
+		Aliases: []string{"n"},
+	})
+
+	c.help = c.set.Help()
+}
+
+// validateFlags checks the command line flags and values for errors.
+func (c *StatsCommand) validateFlags() error {
+	if len(c.set.Args()) > 0 {
+		return errors.New("should have no non-flag arguments")
+	}
+	return nil
+}
+
+func (c *StatsCommand) Run(args []string) int {
+	c.once.Do(c.init)
+
+	if err := c.parseFlags(args); err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		c.UI.Output("\n" + c.Help())
+		return 1
+	}
+
+	if err := c.validateFlags(); err != nil {
+		c.UI.Output(err.Error())
+		return 1
+	}
+
+	if c.flagPod == "" {
+		c.UI.Output("pod name is required")
+		return 1
+	}
+
+	// Setup logger to stream Helm library logs.
+	var uiLogger = func(s string, args ...interface{}) {
+		logMsg := fmt.Sprintf(s, args...)
+		c.UI.Output(logMsg, terminal.WithLibraryStyle())
+	}
+
+	// helmCLI.New() will create a settings object which is used by the Helm Go SDK calls.
+	settings := helmCLI.New()
+	if c.flagKubeConfig != "" {
+		settings.KubeConfig = c.flagKubeConfig
+	}
+	if c.flagKubeContext != "" {
+		settings.KubeContext = c.flagKubeContext
+	}
+
+	if c.flagNamespace == "" {
+		_, _, namespace, err := c.helmActionsRunner.CheckForInstallations(&helm.CheckForInstallationsOptions{
+			Settings:    settings,
+			ReleaseName: common.DefaultReleaseName,
+			DebugLog:    uiLogger,
+		})
+		if err != nil {
+			c.UI.Output("error getting namespace from kubeconfig %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+		c.flagNamespace = namespace
+	}
```

Comment:
This is not quite the namespace you want to grab. This will get the namespace that Consul was installed into. This is generally `consul`, but we advise against users deploying their applications in the `consul` namespace. 

Instead, you will want to grab the namespace currently associated with the users Kubeconfig.

```go
    if c.flagNamespace == "" {
		c.flagNamespace = settings.Namespace()
	}
```